### DISCUSSION TITLE
Store/retreive from database

### DISCUSSION THREAD

**Discussion by rolandow** [angularsen/UnitsNet#1154] [General]
I am trying to figure out how to store and retreive an IQuantity to a database. So basically I want to store a Quantity field and Unit as a string. When I inspect the JSON converter it seems that the string field is exploded, converted to a type, etc. etc.

I was wondering if this is really the way to go. For such a great library surely I must have missed something. I was expecting I could get the full string from QuantityInfo object, and then use that to parse it back to an IQuantity.

**Comment by angularsen** [comment]
This area could definitely be improved.
You can take inspiration from the JSON converters, but generally I would advise to take full control of how you serialize and deserialize values to be tolerant to breaking changes.

For example, one approach would be to serialize 100 cm as `Value=100, Unit="Centimeter", Quantity="Length"`. These names are fairly rarely changed and easily obtained from unit enum types and quantity types.
However, you may need to handle breaking changes at some point in the future, if the library decides to rename "Centimeters" to "Centimetres" or similar. 


Anyway, there is currently no built-in public way to parse these values.
The closest I can think of is `UnitConverter.ConvertByName()`:
https://github.com/angularsen/UnitsNet/blob/ebd3275b84991f5d90113a03fb569cddd655c86e/UnitsNet/UnitConverter.cs#L337-L357

We could probably provide something similar, like `Quantity.ParseByName("Length", "Centimeter", 100)`.
If you are interested in attempting a pull request, I'm happy to assist.
